jepsen/src/jepsen/nemesis/combined.clj

```diff
@@ -234,12 +234,20 @@
                                :strobe-clock          :strobe
                                :bump-clock            :bump}
                               (nt/clock-nemesis)})
-          gen     (->> (nt/clock-gen)
-                       (gen/f-map {:reset          :reset-clock
-                                   :check-offsets  :check-clock-offsets
-                                   :strobe         :strobe-clock
-                                   :bump           :bump-clock})
-                       (gen/delay (:interval opts default-interval)))]
+          db (:db opts)
+          target-specs (:targets (:clock opts) (node-specs db))
+          targets (fn [test] (db-nodes test db
+                                       (some-> target-specs seq rand-nth)))
+          clock-gen (gen/phases
+                     (gen/once {:type :info, :f :check-offsets})
+                     (gen/mix ((juxt reset-gen-select bump-gen-select strobe-gen-select)
```

Comment:
This took me about 15 minutes to figure out, haha. I thought that -select was somehow acting as a verb, but it's actually a function which yields generators. I think I get it now. Comments might be helpful.

Reply:
Simplified and added comment.